server/topic.go

```diff
@@ -46,6 +47,19 @@ type Topic struct {
 	isProxy bool
 	// Name of the master node for this topic if isProxy is true.
 	masterNode string
+	// Topic runs a goroutine (clusterWriteLoop) that reads events from all proxy
+	// multiplexing sessions.
+	// List of proxied sessions.
+	proxiedSessions []*Session
+	// Proxied sessions' channels for the use in the topic's clusterWriteLoop:
+	// i-th session's channels (proxiedSessions[i]) are found at:
+	// proxiedChannels[i * 3 + 1] - send
+	// proxiedChannels[i * 3 + 2] - stop
+	// proxiedChannels[i * 3 + 3] - detach
+	//
+	// proxiedChannels[0] is a special-purpose channel necessary for interrupting
+	// clusterWriteLoop when sessions are added or removed.
+	proxiedChannels []reflect.SelectCase
```

Comment:
I don't understand why Reflect. Why not just use a struct? It makes things hard to read without adding clear value.

Reply:
Ah, I understand now. You want to use a single select to check all channels in one place. Interesting.

Reply:
Correct.